### ISSUE TITLE
Nodes are not reconnecting to each other after OnTerminated due to long time without connection

### ISSUE THREAD

**Issue by vncoelho** [neo-project/neo#462]
@erikzhang, @jsolman 

We are not expert in this part of the code, but we are now with a dedicated team checking it out in different manners.

After removing the connection from a node, as expected, it finished all its connections after some couple of minutes:

```cs
OnTerminated bye bye  
 endPoint.Address=172.18.0.2
OnTerminated bye bye  
 endPoint.Address=172.18.0.6
OnTerminated bye bye  
 endPoint.Address=172.18.0.3
OnTerminated bye bye  
 endPoint.Address=172.18.0.9
OnTerminated bye bye  
 endPoint.Address=172.18.0.8
OnTerminated bye bye  
 endPoint.Address=172.18.0.7
OnTerminated bye bye  
 endPoint.Address=172.18.0.4
```

However, the problem is that it is not being able to fully reconnect to these nodes. This is one of the main causes of chaynsync problems both for Consensus Nodes and for normal Seed RPC nodes.

After all these aforementioned batch of `OnTerminated` calls the actors seams to have some problem in reestablishing the connections, which are often reporting `OnTerminated` after some time.
But is keep reporting `ConnectedPeers.Count` close to the limit (10 as default).

**Comment by vncoelho** [comment]
Even in normal operation connections are dropping all time and calling `OnTerminated` for some peers, even when they are stable, maybe is due to the lack of communication between them.

We are going to double check that.

**Comment by erikzhang** [comment]
Please find out where the problem lies.

**Comment by vncoelho** [comment]
I am trying, master. aehuaheuaea

We noticed that this Timeout is kind of normal.
We need some more experiments. Let's keep this in mind that there is a possible problem. We are not 100% sure.

**Comment by vncoelho** [comment]
Apparently, the lines are these one, @erikzhang:

https://github.com/neo-project/neo/blob/60a02eedc156a66e3955340f0024026f32d296f6/neo/Network/UPnP.cs#L33-L35

**Comment by vncoelho** [comment]
@erikzhang, by commenting these lines everything kept working normally here for me at the nodes that was stuck was able to sync!

**Comment by vncoelho** [comment]
By the way, what is this Socket, Erik? aheuaheuahea

**Comment by vncoelho** [comment]
@erikzhang and @jsolman, the strange think is that when nodes are stuck like this and trying to sync restarting is the best choice.

I mean, by killing the application and restarting again the consensus nodes sync very fast.
In this sense, I think that the problem might be something related to priority in receiving the messages.

The two options I thought are:

* Create a method that kills neosystem and restart everything inside neo
* Stop all services and give total priority in just receiving blocks

What do you think, @erikzhang?

**Comment by vncoelho** [comment]
I like the idea of destroying NeoSystem and initialize it again.

**Comment by vncoelho** [comment]
I believe this has been solved, but let's keep a reference of this on #620.
